LibreNMS/RRD/RrdDefinition.php

```diff
@@ -50,13 +51,20 @@ public static function make()
      * @param int $max Maximum allowed value.  null means undefined.
      * @param int $heartbeat Heartbeat for this dataset. Uses the global setting if null.
      * @return $this
+     * @throws InvalidRrdNameException
      */
     public function addDataset($name, $type, $min = null, $max = null, $heartbeat = null)
     {
         global $config;
+        $name = $this->escapeName($name);
+
+        if (strlen($name) == 0 || strlen($name) > 19) {
```

Comment:
The 0 length checks is fine but > 19 isn't as escapeName returns `substr($name, 0, 19);` so it will never match.